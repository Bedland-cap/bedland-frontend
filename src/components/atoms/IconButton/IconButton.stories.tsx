import { Meta, StoryObj } from '@storybook/react';
import ThemeProvider from 'theme/ThemeContext';
import GlobalStyle from 'theme/globalStyles';
import { colorNames } from 'theme/theme.types';
import IconButton from './IconButton';
import { iconsNames } from '../Icon/icon.types';

const meta = {
  title: 'Molecules/IconButton',
  component: IconButton,
  tags: ['autodocs'],
  args: {
    icon: 'arrowRight',
    color: 'default',
    size: 40,
  },
  argTypes: {
    icon: {
      options: iconsNames,
      description: 'Icon name',
    },
    size: {
      description: 'Icon size',
      defaultValue: 16,
      table: {
        defaultValue: { summary: '16' },
      },
      control: { type: 'range', min: 16, max: 68, step: 2 },
    },
    color: {
      description: 'Color of the icon',
      options: colorNames,
      defaultValue: 'default',
      control: 'select',
      table: {
        type: {
          summary: colorNames.toString().replaceAll(',', ' | '),
        },
        defaultValue: { summary: 'default' },
      },
    },
    onClick: {
      description: 'Function called on click',
      action: 'clicked',
      table: {
        disable: true,
      },
    },
  },
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
  parameters: {
    backgrounds: { default: 'light' },
  },
} satisfies Meta<typeof IconButton>;

export default meta;

type Story = StoryObj<typeof meta>;

export const ArrowRightIconButton: Story = {
  args: {
    icon: 'arrowRight',
    color: 'primaryDark',
    onClick: () => null,
  },
};
