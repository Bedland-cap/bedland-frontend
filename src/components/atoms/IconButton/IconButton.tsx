import Icon from 'components/atoms/Icon/Icon';
import { MouseEventHandler } from 'react';
import { ColorNames } from 'theme/theme.types';
import * as Styled from './iconButton.styled';
import { IconName } from '../Icon/icon.types';

export type IconButtonProps = {
  icon: IconName;
  color: ColorNames;
  size: number;
  onClick: MouseEventHandler<HTMLButtonElement>;
};

const IconButton = ({ icon, color, size, onClick }: IconButtonProps) => (
  <Styled.StyledButton type='button' onClick={onClick}>
    <Icon name={icon} size={size} color={color} isActive={false} />
  </Styled.StyledButton>
);

export default IconButton;
