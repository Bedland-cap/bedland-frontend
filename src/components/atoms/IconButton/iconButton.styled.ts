import styled from 'styled-components';

export const StyledButton = styled.button`
  align-items: center;
  background-color: transparent;
  display: flex;
  justify-content: center;
`;

export default StyledButton;
