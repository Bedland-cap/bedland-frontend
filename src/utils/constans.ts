export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
