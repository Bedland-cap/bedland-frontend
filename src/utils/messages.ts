const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
};

export default ERROR_MESSAGES;
