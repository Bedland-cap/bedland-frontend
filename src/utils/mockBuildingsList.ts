import { Building } from 'store/reducers/building/building_slice';

const mockBuildingsList: Building[] = [
  {
    id: '1',
    name: 'Berry',
    address: 'Berry Street 21',
    managerId: '1',
    floors: '5',
  },
  {
    id: '2',
    name: 'Miedziana',
    address: 'Miedziana Street 3',
    managerId: '1',
    floors: '7',
  },
  {
    id: '3',
    name: 'Seaport',
    address: 'Seaport Avenue 51',
    managerId: '2',
    floors: '5',
  },
  {
    id: '4',
    name: 'Desert',
    address: 'Desert Street 2X',
    managerId: '1',
    floors: '6',
  },
  {
    id: '5',
    name: 'Lakewood',
    address: 'Lakewood Avenue 57',
    managerId: '3',
    floors: '10',
  },
];

export default mockBuildingsList;
