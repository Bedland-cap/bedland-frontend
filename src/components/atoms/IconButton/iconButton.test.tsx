import { renderWithProviders } from 'test.utils';
import IconButton, { IconButtonProps } from './IconButton';

describe('Icon', () => {
  const item = ({ icon, size, color, onClick }: IconButtonProps) =>
    renderWithProviders(<IconButton icon={icon} size={size} color={color} onClick={onClick} />, {});

  it('should renders correctly arrowRight IconButton', () => {
    expect(
      item({
        icon: 'arrowRight',
        size: 20,
        color: 'primaryLight',
        onClick: () => null,
      }),
    ).toMatchSnapshot();
  });
});
