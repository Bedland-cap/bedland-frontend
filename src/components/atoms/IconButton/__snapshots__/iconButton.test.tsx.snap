// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icon > should renders correctly arrowRight IconButton 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<body>
    <div>
      <button
        class="c0"
        type="button"
      >
        <svg
          fill="none"
          height="20"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            id="mask-arrowRight"
          />
          <path
            d="M8.33325 5.83334L12.4999 10L8.33325 14.1667"
            stroke="#919AFF"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.39167"
          />
        </svg>
      </button>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div>
    <button
      class="c0"
      type="button"
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <mask
          id="mask-arrowRight"
        />
        <path
          d="M8.33325 5.83334L12.4999 10L8.33325 14.1667"
          stroke="#919AFF"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.39167"
        />
      </svg>
    </button>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
